pkg/controller/controller.go

```diff
@@ -30,6 +31,8 @@ type AppGwIngressController struct {
 
 	recorder record.EventRecorder
 
+	agicPod *v1.Pod
```

Comment:
Should this be a pointer? If it is we have to deal with the `nil` case.
Is there a situation where we would not be able to retrieve the pod from the pod itself?  Is this a situation, in which we should be allowed to continue running the ingress controller?  It seems to me this will be a situation in which we want to halt and ask for correction of some kind.

Reply:
yes, running on local machine may not have a POD.